src/libraries/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -0,0 +1,1241 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+
+namespace System.Threading
+{
+    /// <summary>
+    /// Provides a way to get mutual exclusion in regions of code between different threads. A lock may be held by one thread at
+    /// a time.
+    /// </summary>
+    /// <remarks>
+    /// Threads that cannot immediately enter the lock may wait for the lock to be exited or until a specified timeout. A thread
+    /// that holds a lock may enter the lock repeatedly without exiting it, such as recursively, in which case the thread should
+    /// eventually exit the lock the same number of times to fully exit the lock and allow other threads to enter the lock.
+    /// </remarks>
+    [Runtime.Versioning.RequiresPreviewFeatures]
+    public sealed partial class Lock
+    {
+        private const int DefaultMaxSpinCount = 22;
+        private const int DefaultAdaptiveSpinPeriod = 100;
+        private const int SpinSleep0Threshold = 10;
+        private const uint MaxDurationMsForPreemptingWaiters = 100;
+
+        private static long s_contentionCount;
+
+        // The field's type is not ThreadId to try to retain the relative order of fields of intrinsic types. The type system
+        // appears to place struct fields after fields of other types, in which case there can be a greater chance that
+        // _owningThreadId is not in the same cache line as _state.
+#if TARGET_OSX && !NATIVEAOT
+        private ulong _owningThreadId;
+#else
+        private uint _owningThreadId;
+#endif
+
+        private uint _state; // see State for layout
+        private ushort _recursionCount;
+        private int _spinCount;
+        private int _waiterStartTimeMs;
+        private AutoResetEvent? _waitEvent;
```

Comment:
Current Lock fields occupy 24 bytes on 64bit. These will be rounded up to 32 bytes.

Perhaps some `int` fields can be switched to `ushort` to make the instance size not to increase?